app/models/job_invocation_composer.rb

```diff
@@ -0,0 +1,217 @@
+class JobInvocationComposer
+  attr_accessor :params, :job_invocation, :host_ids, :search_query
+  attr_reader :job_template_ids
+
+  # TODO aka initialize_from_params
+  def initialize(job_invocation, params)
```

Comment:
since this class is taking in params, would it make more sense for this to live under helpers or controllers?  I can see an argument either way, just wanted to bring it up.